Source/WTF/wtf/text/WTFString.cpp

```diff
@@ -381,35 +381,29 @@ CString String::ascii() const
     // Printable ASCII characters 32..127 and the null character are
     // preserved, characters outside of this range are converted to '?'.
 
-    unsigned length = this->length();
-    if (!length) { 
+    if (isEmpty()) {
         char* characterBuffer;
-        return CString::newUninitialized(length, characterBuffer);
+        return CString::newUninitialized(0, characterBuffer);
     }
 
     if (this->is8Bit()) {
-        const LChar* characters = this->characters8();
+        auto characters = this->span8();
 
         char* characterBuffer;
-        CString result = CString::newUninitialized(length, characterBuffer);
+        CString result = CString::newUninitialized(characters.size(), characterBuffer);
 
-        for (unsigned i = 0; i < length; ++i) {
-            LChar ch = characters[i];
-            characterBuffer[i] = ch && (ch < 0x20 || ch > 0x7f) ? '?' : ch;
-        }
+        for (auto character : characters)
+            *characterBuffer++ = character && (character < 0x20 || character > 0x7f) ? '?' : character;
 
         return result;        
     }
 
-    const UChar* characters = this->characters16();
-
+    auto characters = span16();
     char* characterBuffer;
-    CString result = CString::newUninitialized(length, characterBuffer);
+    CString result = CString::newUninitialized(characters.size(), characterBuffer);
```

Comment:
Does this chop a `size_t` to an `unsigned`?

Reply:
I guess I should update some functions to take in a `size_t` instead of an `unsigned` and add an assertion that the value fits in an unsigned if they ever store the value in an `unsigned` variable. Let me know if you had something else in mind.

Reply:
newUninitialized::newUninitialized takes in a size_t already so no issue for this particular instance.